corehq/apps/userreports/reports/builder/forms.py

```diff
@@ -633,6 +633,17 @@ def _build_data_source(self):
         tasks.rebuild_indicators.delay(data_source_config._id)
         return data_source_config._id
 
+    def _change_indicators(self, indicators):
```

Comment:
Right now, the `DataSourceBuilder` handles most of the logic about constructing the indicators (or at least that is the intention). I think it might be better if you incorporated the functionality of `_change_indicators()` into the `DataSourceBuilder` itself.

I'm not sure what the best interface is, but one thing that comes to mind is that you could change `DataSourceBuilder.indicators` to a method instead of a property, and that method could take `numeric_columns` as an argument (a list of ids or something like that). Then when you need to check if there were changes or not, you could do
```
matching_datasource.configured_indicators == self.ds_builder.indicators(numeric_columns)
```